File: @HackerRank/Problem Solving/10_grading-students/main.py
"""https://www.hackerrank.com/challenges/grading/problem"""

def gradingStudents(grades):
    def round_grade(grade):
        below_by = 5 - grade % 5
        if grade > 37 and below_by < 3:
            return grade + below_by
        return grade
    return [round_grade(grade) for grade in grades]

File: @HackerRank/Problem Solving/10_grading-students/main.js
/* https://www.hackerrank.com/challenges/grading/problem */

function gradingStudents(grades) {
    const roundGrade = (grade) => {
        let belowBy = 5 - grade % 5;
        if (grade > 37 && belowBy < 3) {
            return grade + belowBy;
        }
        return grade;
    }
    return grades.map(roundGrade);
}
